roles/rsyslog/tasks/main_core.yml

```
@@ -1,11 +1,36 @@
 ---
 # SPDX-License-Identifier: GPL-3.0-only
 
-- name: Clean up old config files when purging configuration
+# if inputs are given and rsyslog_enabled is true, rsyslog.conf will be
+# overwritten, so no need to reinstall package
+- name: Check if package needs to get reinstalled to restore rsyslog config if purging
   when:
     - logging_purge_confs | bool | d(false)
+    - not rsyslog_inputs | bool
+    # - not __rsyslog_enabled | bool
+    - not __logging_is_ostree | d(false)
+  block:
+    # it is assumed that the only packages providing config files that might
+    # be modified are the base packages (but not others like iproute);
+    # if this is not so, then additional packages will need to be added to this check
+    - name: Get status of rsyslog packages
+      command: rpm -V {{ item }}  # noqa command-instead-of-module
+      loop: "{{ __rsyslog_base_packages  | select('search', 'rsyslog') | list }}"
+      register: __rsyslog_package_status
+      failed_when: false
+      changed_when: false
+
+    - name: Reset original confs - determine whether to reinstall logging package
+      set_fact:
+        __reinstall_rsyslog: true
+      when: __rsyslog_package_status.results |
+        rejectattr('stdout', 'match', '^package .* is not installed') |
+        selectattr('stdout', 'search', ' /etc/rsyslog[.]conf($|\n)') |
+        list | length > 0
+
+- name: Reset configuration
```

```suggestion
- name: Reset configuration
  when: __reinstall_rsyslog | d(false) | bool
```
ansible-lint wants the `when` condition to be defined before the `block` for readability